packages/web-app-files/src/views/spaces/GenericSpace.vue

```diff
@@ -393,13 +397,36 @@ export default defineComponent({
           performLoaderTask(true, path, fileId)
         }
       )
+      document
+        .getElementsByClassName('files-view-wrapper')[0]
+        .addEventListener('contextmenu', (event) => {
+          const { target } = event
+          if (
+            (target as HTMLElement).closest('tbody') ||
+            (target as HTMLElement).closest('.oc-tiles-item')
+          ) {
+            return
+          }
+          event.preventDefault()
+          const newEvent = new MouseEvent('contextmenu', event)
+          showContextMenu(newEvent)
+        })
     })
 
     onBeforeUnmount(() => {
       visibilityObserver.disconnect()
       eventBus.unsubscribe('app.files.list.load', loadResourcesEventToken)
     })
 
+    const genericContextMenu = ref(null)
+    const showContextMenu = (event) => {
```

Comment:
someone mentioned it would be better to do all this logic in one component which i am not a fan of since this logic toggles the whole component